src/solvers/eigsolve.jl

```diff
@@ -0,0 +1,36 @@
+function eigsolve_updater(
+  init;
+  state!,
+  projected_operator!,
+  outputlevel,
+  which_sweep,
+  sweep_plan,
+  which_region_update,
+  region_kwargs,
+  updater_kwargs,
+)
+  default_updater_kwargs = (;
+    which_eigenvalue=:SR,
+    ishermitian=true,
+    tol=1e-14,
+    krylovdim=3,
+    maxiter=1,
+    verbosity=0,
+    eager=false,
+  )
+  updater_kwargs = merge(default_updater_kwargs, updater_kwargs)  #last collection has precedence
+  howmany = 1
+  which, updater_kwargs = _pop_which_eigenvalue(; updater_kwargs...)
```

Comment:
```suggestion
  (; which) = updater_kwargs
  updater_kwargs = Base.structdiff(updater_kwargs, (; which=nothing))
```
instead of defining `_pop_which_eigenvalue`. I'd be ok with separating out a separate function if it was more general (like handled popping out any specified entry of a `NamedTuple`, not just `which`), but as it is `_pop_which_eigenvalue` is very specialized.

Reply:
Sure, i wasn't aware of how to do it with `structdiff`.